Sources/StringsLintFramework/Parsers/StructuredPlaceholderCommentParser.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  StructuredPlaceholderCommentParser.swift
+//  
+//
+//  Created by Mark Hall on 2022-07-19.
+//
+
+import Foundation
+
+public struct StructuredPlaceholderCommentParser: LocalizableParser {
+
+    enum StructuredPlaceholderCommentParserError: Error {
+        case missingHeaderSeparatorComment
+    }
+
+    public static var identifier: String {
+        return "structured_placeholder_comment_parser"
+    }
+
+    public var supportedFileExtentions: [String] {
+        return [ "strings" ]
+    }
+
+    public init() {
+    }
+
+    public init(configuration: Any) throws {
+        // Parser does not support any configuration
+        self.init()
+    }
+
+    //TODO: (Mark Hall, July 18) only need to support the english file
```

Comment:
did you figure out this TODO?

Reply:
This should probably be a condition in the rule itself, not the parser

Reply:
The way the existing code works, the parser is the one that defines which files it supports and the rules just use parsers. Since I am using the same rule for both the strings and stringsdict parser to enforce this new JSON comment format, I'd have to add the logic to only lint the english files to both parsers. I think the better way to do this is what I mentioned above, which is to put only the path to the english files into the `.stringslint.yml` file so it will only ever look for the english files for both `.strings` and `.stringsdict` files.